### ISSUE TITLE
Metrics calculation on F1-macro and accuracy

### ISSUE THREAD

**Issue by junhua** [flairNLP/flair#1558]
Does Flair's metrics calculation use a different scheme than sklearn.metrics for f1-macro and accuracy? 

I realised that for all lstm models ran with Flair has a higher f1-macro than accuracy, and all other models (LR, SVM, NB, CNN, transformers) that I used sklearn to evaluate have a lower f1-macro. 

It doesn't really look like coincidence - see screenshot below

<img width="760" alt="image" src="https://user-images.githubusercontent.com/3516360/80558861-606aab00-8a0e-11ea-9dde-c5b529de0a7a.png">

**Comment by junhua** [comment]
Maybe this helps better

<img width="1737" alt="image" src="https://user-images.githubusercontent.com/3516360/80563468-d9253380-8a1d-11ea-9d90-3e4b689ffe99.png">

**Comment by djstrong** [comment]
Check empirically.

**Comment by alanakbik** [comment]
Hi @junhua thanks for sharing this. I'll habe to take a closer look. 

Are you working on flair 0.4.5 or on master branch?

**Comment by djstrong** [comment]
The paths in the screenshot does not match.

But Flair calculation of the accuracy per class is wrong (for first class it should be 95.8% - TP+TN/(TP+TN+FP+FN)), the F1 is correct.

**Comment by alanakbik** [comment]
Yes, this should be fixed in master branch.

**Comment by junhua** [comment]
> Hi @junhua thanks for sharing this. I'll have to take a closer look.
> 
> Are you working on flair 0.4.5 or on master branch?

flair 0.4.5.. so it's fixed in master?

**Comment by junhua** [comment]
> paths

Sharp eyes :) I changed the folder name after training for consistency of my own naming convention for this particular example. I can upload  5 other examples that show the differences :)

**Comment by djstrong** [comment]
In master the calculations should be ok. But there are some other problems: probably predictions in test.tsv does not match with Flair reported results. So, we have to check if e.g. tp, fp, fn, tn in test.tsv match numbers in the Flair log.

**Comment by junhua** [comment]
I suppose this could be a severe issue because some researches (like myself) used to take the training log stats and put them in papers..

**Comment by junhua** [comment]
> The paths in the screenshot does not match.
> 
> But Flair calculation of the accuracy per class is wrong (for first class it should be 95.8% - TP+TN/(TP+TN+FP+FN)), the F1 is correct.
> 
> In Flair accuracy is implemented as TP/(TP+FP+FN):
> 
> ```
> (self.get_tp(class_name))
>                 / (
>                     self.get_tp(class_name)
>                     + self.get_fp(class_name)
>                     + self.get_fn(class_name)
>                 ),
> ```

Oops... There:

<img width="1870" alt="image" src="https://user-images.githubusercontent.com/3516360/80687198-85d5e280-8afc-11ea-8a25-8c3e4232bb0e.png">

<img width="1877" alt="image" src="https://user-images.githubusercontent.com/3516360/80687288-a867fb80-8afc-11ea-863d-c2e4f4d283b3.png">

<img width="1661" alt="image" src="https://user-images.githubusercontent.com/3516360/80687406-d4837c80-8afc-11ea-814a-cbd83cbb185a.png">

**Comment by djstrong** [comment]
So it is correct now? It is not from master?

**Comment by junhua** [comment]
Not using master yet - still 0.4.5

**Comment by alanakbik** [comment]
You can install master by doing 
```console
pip install --upgrade git+https://github.com/flairNLP/flair.git
```
can you try if this works for you?
